packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -0,0 +1,3369 @@
+        mocks: [
+          // NetworkController
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x2',
+            },
+          },
+        ],
+      });
+      const { networkController, transactionController } =
+        await newController();
+      const stopTrackinSpy = jest.spyOn(
+        transactionController,
+        'stopTrackingByNetworkClientId',
+      );
+
+      const configurationId =
+        await networkController.upsertNetworkConfiguration(
+          {
+            ...networkClientConfiguration,
+            rpcUrl: 'https://mock.rpc.url',
+          },
+          {
+            setActive: false,
+            referrer: 'https://mock.referrer',
+            source: 'dapp',
+          },
+        );
+
+      networkController.removeNetworkConfiguration(configurationId);
+
+      // advance time to trigger events
+      await advanceTime({ clock, duration: 1000 });
+
+      expect(stopTrackinSpy).toHaveBeenCalledTimes(1);
+      expect(transactionController).toBeDefined();
+    });
+  });
+
+  describe('startIncomingTransactionPolling', () => {
+    // TODO(JL): IncomingTransactionHelper doesn't populate networkClientId on the generated tx object. Should it?..
+    it('should add incoming transactions to state with the correct chainId for the given networkClientId on the next block', async () => {
+      mockNetwork({
+        networkClientConfiguration: mainnetNetworkClientConfiguration,
+        mocks: [
+          // NetworkController
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x2',
+            },
+          },
+        ],
+      });
+
+      const selectedAddress = ETHERSCAN_TRANSACTION_BASE_MOCK.to;
+
+      const { networkController, transactionController } = await newController({
+        getSelectedAddress: () => selectedAddress,
+      });
+
+      const expectedLastFetchedBlockNumbers: Record<string, number> = {};
+      const expectedTransactions: Partial<TransactionMeta>[] = [];
+
+      const networkClients = networkController.getNetworkClientRegistry();
+      // NOTE(JL): This doesn't seem to work for the globally selected provider because of nock getting stacked on mainnet.infura.io twice
```

Comment:
replace all of these comments with `Skip globally selected provider because we can't stack nock requests` or something like that?

Reply:
https://github.com/MetaMask/core/pull/3643/commits/13ed1e8bce76debf9ef9a90f5b29f8cfc13b25f2